py4j-python/release_process.txt

```diff
@@ -1,31 +1,38 @@
 0. Update all versions (version.py, properties, documentation, index.rst).
 1. Write changelog
-2. gradle clean
+2. gradle clean (at py4j/py4j-java)
 
 3. close all bugs on github
 4. tag the release on git
 
-5. gradle buildPython
-6. create signature using gpg (gpg -b --armor)
+5. gradle buildPython  (at py4j/py4j-java)
+6. create signature using gpg (gpg --export --armor YOUR_KEY, also refer to 'Preparing gpg key' in https://spark.apache.org/release-process.html)
 
-7. upload to PyPI using twine
+7. upload to PyPI using twine (at py4j/py4j-python). For example:
 
-8. mvn package javadoc:jar source:jar
-9. create signature for the three jar files and the pom (gpg -b --armor py4j-....jar)
-10. jar -cvf bundle.jar pom.xml ... (all files)
-11. create staging release in central (https://oss.sonatype.org Staging Upload, Upload Bundle)
-12. close release
-13. publish release
+    python setup.py sdist
+    python setup.py bdist_wheel --universal
+    twine upload dist/*
 
-14. gradle updateSite
-15. Extract and modify artifacts.xml and content.xml to change build path to:
+8. make sure you have 'settings.xml' under '~/.m2', see also https://central.sonatype.org/publish/publish-maven/#distribution-management-and-authentication
+9. if you are a new user in Sonatype, you should replace the upload URL in pom.xml (at py4j/py4j-java). For example:
+
+    # In Mac
+    sed -i '' 's/oss.sonatype.org/s01.oss.sonatype.org/g' pom.xml
+    # or in Linux OS
+    sed 's/oss.sonatype.org/s01.oss.sonatype.org/g' pom.xml
+
+10. mvn clean deploy (at py4j/py4j-java)
```

Comment:
I wrote a lot of things here but essentially the release process would be, once it's setup,:

1. For PyPI:

    ```bash
    python setup.py sdist
    python setup.py bdist_wheel --universal
    twine upload dist/*
    ```

2. For Sonatype:
 
    ```bash
    mvn clean deploy
    ```

Reply:
Just to be sure, before setup.py sdist, we would still have to run gradle buildPython before right (because we are shipping the jar with the python distribution)?

Reply:
Ok, I just read the whole file, it works for me! Great improvement, i really no next to nothing about maven :-)

Reply:
FYI, I am rolling another release (see https://github.com/py4j/py4j/pull/475) - the new release broke (unofficial) Python 3.10 support. I am adding it into the CI at https://github.com/py4j/py4j/pull/477, and will roll another release.

Sorry I didn't expect this .. 😢

Reply:
BTW, I think you're correct here about `gradle buildPython`. I will take another look and fix it better when releasing the next one.

Reply:
0.10.9.5 is done up to step 10. identically. Maybe we can skip 0.10.9.4 ..